### PR TITLE
Add configuration file for Travis CI

### PR DESCRIPTION
**Pull request by jschlyter**

### CODE REVIEW

**Comment by jschlyter** [comment]
Now testing both Botan & OpenSSL as well as file & sqlite object stores. Example at https://travis-ci.org/jschlyter/SoftHSMv2.

File: .travis.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	CONF_CRYPTO="$CONF_CRYPTO --with-crypto-backend=openssl --with-openssl=/usr"
+	CONF_CRYPTO="$CONF_CRYPTO --disable-gost"
+	openssl version -a
+	;;
+*)
+	echo "Unknown crypto backend"
+	exit 1
+esac
+
+case $OBJSTORE in
```
Comment (bellgrim) [inline .travis.sh:21]: Maybe we do not need to build with and without "--with-objectstore-backend-db". The file backend will always be supported. It is not like OpenSSL or Botan, where you only have one backend in the build.

Perhaps we always can make the test build with "--with-objectstore-backend-db --with-migrate"? Or do we want to test a build without SQLite support?

File: .travis.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+CONF_CRYPTO=""
+CONF_OBJSTORE=""
+
+case $CRYPTO in
+botan)
+	CONF_CRYPTO="$CONF_CRYPTO --with-crypto-backend=botan --with-botan=/usr"
+	CONF_CRYPTO="$CONF_CRYPTO --disable-ecc --disable-gost"
```
Comment (bellgrim) [inline .travis.sh:9]: Too bad that Travis is using Ubuntu 12.04, which only include Botan 1.10.0. Thus not supporting ECC and GOST.

Comment (jschlyter) [inline .travis.sh:9]: Yes. On the other hand, OpenSSL has ECC so that part is covered.

File: .travis.sh (hunk 1 above)
Comment (jschlyter) [inline .travis.sh:21]: Yes, without SQLite would be useful I believe.

**Comment by bellgrim** [comment]
One more question before merge... I guess that we do not want these files included in the dist?

**Comment by jschlyter** [comment]
Correct, this is repo only.

**Pull request merged**